upcoming: [M3-7874] - Linode Create Refactor - Marketplace App Sections

bnussman-akamai opened the pull request:
## Description 📝

- Adds the `New apps`, `Popular apps`, and `all apps` sections to the new Linode Create flow 🆕 

## Preview 📷

| Before  | After   |
| ------- | ------- |
| ![Screenshot 2024-05-28 at 3 28 15 PM](https://github.com/linode/manager/assets/115251059/e253f7c6-4f75-4390-a543-7b2dad5c855f) | ![Screenshot 2024-05-28 at 3 28 03 PM](https://github.com/linode/manager/assets/115251059/39f49dff-e1b6-49f3-9c47-b3853760eef9) |
| Notice how there are no sections  | It now looks like the old flow with the proper sections  | 

## How to test 🧪

### Prerequisites
- Turn on the Linode Create v2 feature flag using local dev tools

### Verification steps
- Checkout this PR locally or use the internal preview link
- Go to the Linode Create flow
- Got to the marketplace tab
- Verify you see the sections `New apps`, `Popular apps`, and `all apps` 
- Check for parity with production (the old Linode Create flow)
  - Verify the apps are the same under each section

## As an Author I have considered 🤔

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

hkhalil-akamai commented on packages/manager/src/features/OneClickApps/types.ts:
```diff
@@ -5,6 +5,13 @@ export interface OCA {
   colors: Colors;
   description: string;
   href?: string;
+  /**
+   * Set isNew to `true` if you want the app to show up in the "New apps"
+   * section on the Linode Create flow.
+   *
+   * @note this value only affects Linode Create v2
+   */
+  isNew?: boolean;
```
If the new apps are rotated frequently, maybe its better to use a feature flag? (So it's not dependent on our release cycle)

bnussman-akamai replied:
Yeah, this is a good point.  Before we launch v2, I will make a new flag that will allow us to override **anything** for emergencies

hkhalil-akamai reviewed (approved):
Code changes are looking clean 🧼 and found no regressions from prod.

hana-linode reviewed (commented):
Can we also fix the Setup headers for special characters and/or add back the app icon & node notice?

| Prod  | Linode Create v2   |
| ------- | ------- |
| ![image](https://github.com/linode/manager/assets/115299789/33549159-cf31-49ab-a8da-e47fafbaebb4) | ![image](https://github.com/linode/manager/assets/115299789/d23de071-24db-46e6-bf2e-58f1a75a90d7) |

bnussman-akamai commented:
@hana I addressed that feedback!

I also went ahead and deleted the old `oneClickApps` list and the `stackScriptUtils.ts` baseApps list. `oneClickAppsv2` is now the source of truth.

bnussman-akamai commented:
@hana-linode Search is not yet implemented, it will come in a later PR
